python_modules/dagster-graphql/dagster_graphql/implementation/fetch_runs.py

```diff
@@ -609,12 +609,18 @@ def get_runs_feed_entries(
     else:
         backfills = []
 
-    runs = [
-        GrapheneRun(run)
-        for run in instance.get_run_records(
-            limit=fetch_limit, cursor=runs_feed_cursor.run_cursor, filters=run_filters
-        )
-    ]
+    # if we are not showing runs within backfills and the backfill_id filter is set, we know
+    # there will be no results, so we can skip fetching runs
+    should_fetch_runs = not (exclude_subruns and run_filters.tags.get(BACKFILL_ID_TAG) is not None)
```

Comment:
when we add the option to only show backfills this condition will become more complicated and probably warrant it's own function